From 3ef080834b4714370201da5285221d6b0208563f Mon Sep 17 00:00:00 2001
From: Florian Boor <florian.boor@kernelconcepts.de>
Date: Tue, 19 Feb 2019 16:57:49 +0100
Subject: [PATCH] merge set-enet_ref_clk-to-50-mhz.patch

---
 drivers/clk/imx/clk-imx6q.c | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/drivers/clk/imx/clk-imx6q.c b/drivers/clk/imx/clk-imx6q.c
index 8f477de8a257..916dbfb8696e 100644
--- a/drivers/clk/imx/clk-imx6q.c
+++ b/drivers/clk/imx/clk-imx6q.c
@@ -1063,6 +1063,20 @@ static void __init imx6q_clocks_init(struct device_node *ccm_node)
        // This clock may be needed very early on
        imx_clk_prepare_enable(clk[IMX6QDL_CLK_ENET_REF]);
 
+       np = of_find_compatible_node(NULL, NULL, "fsl,imx6q-fec");
+       if (np) {
+               u32 clock_frequency;
+               int ret;
+
+               ret = of_property_read_u32(np, "ref-clock", &clock_frequency);
+               if (ret == 0) {
+                       printk("fec ref-clock: %d\n", clock_frequency);
+                       imx_clk_set_rate(clk[IMX6QDL_CLK_ENET_REF], clock_frequency);
+               }
+       }
+       // This clock may be needed very early on
+       imx_clk_prepare_enable(clk[IMX6QDL_CLK_ENET_REF]);
+
 #ifdef CONFIG_MX6_VPU_352M
 	/*
 	 * If VPU 352M is enabled, then PLL2_PDF2 need to be
