From 4c1d2fe6ff8aabe9f82fe6dc6417b22bd150d319 Mon Sep 17 00:00:00 2001
From: Florian Boor <florian.boor@kernelconcepts.de>
Date: Wed, 16 May 2018 16:44:32 +0200
Subject: [PATCH] Disable Werror for default build - useless for all other than
 developers.

---
 board/karo/tx6/config.mk | 1 -
 1 file changed, 1 deletion(-)

diff --git a/board/karo/tx6/config.mk b/board/karo/tx6/config.mk
index ce8e740..e2ac4d7 100644
--- a/board/karo/tx6/config.mk
+++ b/board/karo/tx6/config.mk
@@ -10,7 +10,6 @@ OBJCOPYFLAGS += -j .pad
 LOGO_BMP = logos/karo.bmp
 #PLATFORM_CPPFLAGS += -DDEBUG
 #PLATFORM_CPPFLAGS += -Wno-unused-but-set-variable
-PLATFORM_CPPFLAGS += -Werror
 
 ifneq ($(CONFIG_SECURE_BOOT),)
 	# promote config variable to linker script
-- 
2.1.4

